src/functional/list.h

```diff
@@ -0,0 +1,31 @@
+#ifndef LIST_H
+#define LIST_H
+
+#include <stddef.h>
+#include <stdbool.h>
+
+typedef struct list_t list_t;
+
+struct list_t {
+	void* head;
+	void* tail;
```

Comment:
Your definition of list_t is against the naming convention of linked lists.
Seriously I don't like it. Can we have a more conventional naming like this?

    struct linked_list_node {
    void *element;
    struct linked_list_node *next;
    }